core/common/src/annotation/TextBlock.ts

```diff
@@ -440,13 +456,17 @@ export class TextBlock extends TextBlockComponent {
   public width: number;
   /** The alignment of the document's content. */
   public justification: TextBlockJustification;
+  /** The margins of the document. */
+  public margins: TextBlockMargins;
   /** The ordered list of paragraphs within the document. */
   public readonly paragraphs: Paragraph[];
 
   private constructor(props: TextBlockProps) {
     super(props);
     this.width = props.width ?? 0;
     this.justification = props.justification ?? "left";
+    // Assign default margins if not provided
+    this.margins = { left: 0, right: 0, top: 0, bottom: 0, ...props.margins };
```

Comment:
```suggestion
    this.margins = {
      left: props.margins.left ?? 0,
      right: props.margins.right ?? 0,
      top: props.margins.top ?? 0,
      bottom: props.margins.bottom ?? 0,
    };
```

If `props` is, e.g., `{ top: undefined }` then the spread operator will overwrite your `0` with `undefined` in `this.margins`.

Reply:
Should be fixed now